drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -97,6 +98,14 @@ static int lis2dh_channel_get(const struct device *dev,
 	int ofs_end;
 	int i;
 
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+	(void)pm_device_state_get(dev, &state);
+	/* Do not allow channel get from suspended state */
+	if (state == PM_DEVICE_STATE_SUSPENDED)
+		return -EIO;
+#endif
```

Comment:
please avoid doing this. Instead, use PM runtime APIs, so that you make sure sensor is always ON when sampling.

Reply:
@gmarull I copied this approach from what I saw in other sensor drivers. I see this pattern in most of the drivers that support power management. It sounds like we're moving away from that? Is there an existing driver that I should pattern my approach on?

Reply:
@gmarull Do you mean that I should remove this (and the other) check, that the application is responsible for using the runtime calls to suspend/resume before using the device?

Reply:
Yes, application is responsible to make sure a device is awake when using it. Another option, is to use the PM device runtime API so that sensor itself takes care of suspending/resuming as needed.

Reply:
@gmarull It seems to me like one way or the other should be standard to make things behave consistently. Looking at other drivers, it looks like we're leaving this to the application in nearly all cases. So, I've removed these checks.